test/mocks/grpc/mocks.h

```diff
@@ -90,6 +90,7 @@ class MockAsyncClient : public RawAsyncClient {
                const Http::AsyncClient::StreamOptions& options));
 
   std::unique_ptr<testing::NiceMock<Grpc::MockAsyncRequest>> async_request_;
+  int send_cnt_{};
```

Comment:
I know due to our IM that you are only putting send_cnt here so that TSAN can detect a race on it if it's not properly threaded.

But you need to comment this strategy in the code, so a future reader of this code doesn't see this as a write-only variable and just remove it.

Reply:
Also you should probably test send_cnt_ in the tests against an expected value, at the end.

Reply:
Thanks, the send_cnt_ is now tested.

Reply:
+1 for a comment here
(was about to ask why it's needed)